codegenerator/cli/templates/dynamic/codegen/src/IO.res.hbs

```diff
@@ -202,7 +226,10 @@ let executeBatch = async (sql, ~inMemoryStore: InMemoryStore.t, ~isInReorgThresh
         set{{entity.name.capitalized}}s,
         {{/each}}
       ],
-    )->Belt.Array.map(dbFunc => sql->dbFunc)
+      //History pruning needs to happen last in the transaction
+      //It deletes all unneeded history rows outside of the reorg threshold
+      pruneEntityHistory,
```

Comment:
History now gets pruned in the transaction after new history has been added and entities have been updated.